flit_core/tests_core/test_config.py

```diff
@@ -178,3 +184,45 @@ def test_bad_pep621_readme(readme, err_match):
     }
     with pytest.raises(config.ConfigError, match=err_match):
         config.read_pep621_metadata(proj, samples_dir / 'pep621')
+
+@pytest.mark.parametrize(('value', 'license'), [
+    # Accept any string in project.license.text
+    ({"text": "mit"}, "mit"),
+    ({"text": "Apache Software License"}, "Apache Software License"),
+])
+def test_license_text(value, license):
+    proj = {
+        'name': 'module1', 'version': '1.0', 'description': 'x', 'license': value
+    }
+    info = config.read_pep621_metadata(proj, samples_dir / 'pep621')
```

Comment:
```suggestion
    info = config.read_pep621_metadata(proj, samples_dir / 'pep621' / 'pyproject.toml')
```
`read_pep621_metadata` expects the path to the `pyproject.toml` file, not the dir.